### ISSUE TITLE
centos7_cluster job crashes locally with an error

### ISSUE THREAD

**Issue by rishabh-d-dave** [ceph/ceph-ansible#3421]
**Bug Report**

What happened: Running a  CI job locally for luminous against centos7_scenario leads to an error with the following traceback -

```
The full traceback is:
Traceback (most recent call last):
  File "<stdin>", line 113, in <module>
  File "<stdin>", line 105, in _ansiballz_main
  File "<stdin>", line 48, in invoke_module
  File "/tmp/ansible_ceph_key_payload_SjfPup/__main__.py", line 714, in <module>
  File "/tmp/ansible_ceph_key_payload_SjfPup/__main__.py", line 710, in main
  File "/tmp/ansible_ceph_key_payload_SjfPup/__main__.py", line 656, in run_module
  File "/tmp/ansible_ceph_key_payload_SjfPup/__main__.py", line 475, in lookup_ceph_initial_entities
NameError: global name 'module' is not defined

fatal: [ceph-mon1]: FAILED! => {
    "changed": false, 
    "rc": 1
}

MSG:

MODULE FAILURE
See stdout/stderr for the exact error


MODULE_STDERR:

Traceback (most recent call last):
  File "<stdin>", line 113, in <module>
  File "<stdin>", line 105, in _ansiballz_main
  File "<stdin>", line 48, in invoke_module
  File "/tmp/ansible_ceph_key_payload_SjfPup/__main__.py", line 714, in <module>
  File "/tmp/ansible_ceph_key_payload_SjfPup/__main__.py", line 710, in main
  File "/tmp/ansible_ceph_key_payload_SjfPup/__main__.py", line 656, in run_module
  File "/tmp/ansible_ceph_key_payload_SjfPup/__main__.py", line 475, in lookup_ceph_initial_entities
NameError: global name 'module' is not defined
```

What you expected to happen: The job should've completed running successfully.

How to reproduce it (minimal and precise): Running following command with repo root as CWD should be enough - `sudo ANSIBLE_FORCE_COLOR=TRUE RELEASE=luminous SCENARIO=centos7_cluster WORKSPACE=./ bash -x ./ceph-ansible-prs.sh | tee ~/centos7_cluster.log`

**Comment by rishabh-d-dave** [comment]
I could reproduce this issue on my own machine as well as on a machine from Sepia lab. Right now, I am trying to fix it by adding custom debug message to `library/ceph_keys.py` and by trying to find the commit that leads to this error since `luminous-centos7_cluster` completed running successfully on an unupdated repo.

**Comment by noahdesu** [comment]
@rishabh-d-dave see the referenced PR. It looks like the undefined variable error is actually masking another error. So I would expect that when you re-run the tests you'll see another error.

**Comment by rishabh-d-dave** [comment]
@noahdesu You are right I did see a different error. I saw error the error `'auth_dump' key not present in json output`, mostly emanated from [this line](https://github.com/ceph/ceph-ansible/blob/master/library/ceph_key.py#L481).

Following command gets executed [at line #564 in ceph_key.py](https://github.com/ceph/ceph-ansible/blob/master/library/ceph_key.py#L564) `sudo ceph -n mon. -k /var/lib/ceph/mon/ceph-ceph-mon1/keyring --cluster ceph auth ls -f json` which returns an empty JSON value (i.e. `{}`). And when the execution proceeds, the expected keys turn out to be absent, which leads to the error mentioned above.

**Comment by rishabh-d-dave** [comment]
@noahdesu master branch

**Comment by rishabh-d-dave** [comment]
@leseb The root cause (reported [here](https://github.com/ceph/ceph-ansible/issues/3421#issuecomment-445274382)) behind this issue is not yet fixed.

The error mentioned in my original report is replaced by the following one -

```

<ceph-mon1> (1, '\n{"msg": "\'auth_dump\' key not present in json output:", "failed": true, "rc": 1, "invocation": {"module_args": {"directory_mode": null, "force": null, "remote_src": null, "cluster": "ceph", "import_key": true, "owner": "ceph", "follow": false, "group": "ceph", "unsafe_writes": null, "secret": null, "content": null, "serole": null, "setype": null, "state": "fetch_initial_keys", "dest": "/etc/ceph/", "selevel": null, "caps": null, "regexp": null, "src": null, "name": null, "seuser": null, "delimiter": null, "mode": "0400", "attributes": null, "backup": null}}}\n', '')
fatal: [ceph-mon1]: FAILED! => {
    "changed": false, 
    "invocation": {
        "module_args": {
            "attributes": null, 
            "backup": null, 
            "caps": null, 
            "cluster": "ceph", 
            "content": null, 
            "delimiter": null, 
            "dest": "/etc/ceph/", 
            "directory_mode": null, 
            "follow": false, 
            "force": null, 
            "group": "ceph", 
            "import_key": true, 
            "mode": "0400", 
            "name": null, 
            "owner": "ceph", 
            "regexp": null, 
            "remote_src": null, 
            "secret": null, 
            "selevel": null, 
            "serole": null, 
            "setype": null, 
            "seuser": null, 
            "src": null, 
            "state": "fetch_initial_keys", 
            "unsafe_writes": null
        }
    }, 
    "rc": 1
}

MSG:

'auth_dump' key not present in json output:
```

**Comment by rishabh-d-dave** [comment]
Would it be more appropriate to open a new issue with this error?

**Comment by leseb** [comment]
If `auth_dump` is not present, this means there is something wrong with your cluster, make sure it's healthy.

**Comment by noahdesu** [comment]
I don't think the master branch is intended to work with luminous, but I may be wrong when it comes to functional tests. Despite tox _letting_ you run them, I had some issues with this configuration that were resolved when I either switched to stable-3.2 or run with `dev-xxx` in master. @leseb confirm?
